--- gcc-4.0.4/gcc/collect2.c	2005-06-06 21:21:03.000000000 +0200
+++ gcc-4.0.4_patched/gcc/collect2.c	2017-09-28 12:01:26.575046126 +0200
@@ -1577,7 +1577,7 @@
   if (redir)
     {
       /* Open response file.  */
-      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT);
+      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH);
 
       /* Duplicate the stdout and stderr file handles
 	 so they can be restored later.  */
